src/js/parsers/captions/dfxp.ts

```diff
@@ -6,7 +6,7 @@ import type { CaptionEntryData } from './captions.types';
 
 type ParagraphNodes = HTMLCollectionOf<Element & { text?: string }>;
 
-export default function Dfxp(xmlDoc: XMLDocument): CaptionEntryData[] {
+export default function Dfxp(xmlDoc: XMLDocument | null): CaptionEntryData[] {
```

Comment:
Do we actually call `Dfxp(null)` anywhere? We shouldn't allow that. Adding `null` as an accepted type removes protection from calls that would pass in null.

The `if (!xmlDoc) {` guard is in case it's undefined, but that doesn't mean we should let TypeScript allow undefined or null arguments.

Reply:
![image](https://user-images.githubusercontent.com/2903955/82229014-de2c2300-98f7-11ea-9a4e-f57d22481a31.png)

Reply:
![image](https://user-images.githubusercontent.com/2903955/82229063-f13ef300-98f7-11ea-991e-fb80822e10ac.png)

Reply:
Added a check here to throw an error if responseXML is null

Reply:
I don't see it https://github.com/jwplayer/jwplayer/pull/3667/files#diff-6bb2a789e01fbaf7e5d02917600e01c9L58